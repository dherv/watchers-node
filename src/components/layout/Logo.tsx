import React from "react";
import styled from "styled-components";

const Logo = () => <Heading>watchers</Heading>;

const Heading = styled.h1`
  font-size: 2rem;
  font-weight: 600;
  text-transform: capitalize;
`;

export default Logo;
