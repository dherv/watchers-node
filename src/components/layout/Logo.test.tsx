import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Logo, { Heading } from "./Logo";

describe("Logo component", () => {
  const wrapper = shallow(<Logo />);
  describe("user interface", () => {
    test("should display one Heading component", () => {
      expect(wrapper.find(Heading).text()).toEqual("watchers");
    });
  });

  describe("snapshot", () => {
    test("should match", () => {
      expect.assertions(1);
      const wrapper = renderer.create(<Logo />).toJSON();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
