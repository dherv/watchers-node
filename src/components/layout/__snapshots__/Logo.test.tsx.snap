// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component snapshot should match 1`] = `
.c0 {
  font-size: 2rem;
  font-weight: 600;
  text-transform: capitalize;
}

<h1
  className="c0"
>
  watchers
</h1>
`;
